api/src/modules/propositions/propositions.handler.ts

```diff
@@ -0,0 +1,144 @@
+import type { FastifyReply as Reply, FastifyRequest } from "fastify";
+import type { proposition } from "@src/types/proposition";
+import type Filters from "@src/types/Filters";
+import filtersFactorySlots from "@src/utils/filtersFactorySlots";
+
+type Request = FastifyRequest<{
+  Querystring: {
+    filter: Filters.Slot;
+  };
+  Params: {
+    id: number;
+  };
+}>;
+
+export const handleGetAllSlots = async (request: Request, reply: Reply) => {
+  const { prisma } = request;
+  const filters = filtersFactorySlots(request.query.filter);
+
+  try {
+    const slots = await prisma.proposition_slot.findMany(
+      filters && {
+        where: { AND: [...filters] },
+      }
+    );
+
+    if (slots.length === 0) {
+      reply.code(404);
+      throw new Error("Aucun créneau disponible.");
+    }
+
+    reply.send(slots);
+  } catch (error) {
+    reply.send(error);
+  }
+}
+
+export const handleBookSlot = async (request: Request, reply: Reply) => {
```

Comment:
Fusionner book et unbook en un seul handler ?

Reply:
Please comunicate using English only. the `book` function aim to let any registered user to subscribe to a slot. `unBook` is for Admin only.

Reply:
Yeah I got this but the two code blocks are redundants. Maybe we can just factorize the mechanic in a function and call it as you did with two different endpoints ?